March 2023 newsletter - Open for ideas (#208)
readthedocs/blog: newsletter-march-2023.rst

.. _rclone: https://rclone.org/

Upcoming features
-----------------

..
  Notes:

  Next newsletter:
  Make a general announcement of our Roadmap

  General:

  When creating newsletter drafts, we keep the items here from the previous newsletter.
  This is in order to ensure due follow-up on features that are announced publicly.
  
  Feature done? A great follow-up is to add what was previously an upcoming feature as a released feature in the former section.
  
  Feature not done?
  Make sure that upcoming features are announced with a link to issues or PRs where the progress can be seen.
  If this is done, then subsequent newsletters aren't compelled to share progress when it's uninteresting.
  
  If a feature was announced as upcoming but isn't yet released,
  then try rephrasing the announcement as a general news update about the progress and where it can be followed.

- 🪄️ The Django application that handles uncached requests on all our hosted documentation is called El Proxito.
  And that application is getting a major refactor,
  improving both speed and features.
  Most of the refactoring work is already merged and within the coming weeks,

>   we will start to roll it out as a kind of A/B testing.

Author: ```suggestion
  we will start to roll it out gradually.
```

This should be better... A/B testing isn't it.

Reviewer: Yea, this is better 👍

.. post:: March 1, 2023
   :tags: newsletter, python
   :author: Ben
   :location: MLM

.. meta::
   :description lang=en:
      Company updates and new features from the last month,
      current focus, and upcoming features.

Read the Docs newsletter - March 2023
=====================================

News and updates
----------------

> Here are the latest updates from our team since the :doc:`previous newsletter </newsletter-february-2023>`:

Reviewer: I wonder about the value of this? I think we can probably get rid of it, since it shows up in the opengraph preview, and doesn't provide value.

Author: Judging from how the meta description is inserted before everything else, it will read better without this, yes :+1:

- ⭐️ We passed our 10,000th issue/pull request on GitHub. And it's pretty much an equal split between the `5039 issues <https://github.com/readthedocs/readthedocs.org/issues>`__ and `4872 pull requests <https://github.com/readthedocs/readthedocs.org/pulls>`__ now registered.
  Thanks to the whole community for building this together through code, issues, suggestions... and documentation!
- 🌪️ Follow up: Build times have gone rapidly down after last month's introduction of parallel uploading of artifacts with `rclone`_.

>   Depending on the number of files in a build outputs,

Reviewer: ```suggestion
  Depending on the number of files in your build output,
```

  build times may have gone down several seconds or several minutes.
  For instance, a large project like `Write the Docs <https://readthedocs.org/projects/writethedocs-www/builds/>`__ has gone down from ~7 minutes to under 3 minutes.
  If you want to see the change for your project,
  have a look at your build times before and after **February 8**.

> - 1️⃣️ We are going to have 1 main Read the Docs website in the future.
>   The first move has been done now,
>   as readthedocs.com now points to `about.readthedocs.com <https://about.readthedocs.com>`__ instead of being a separate site for Read the Docs for Business.
>   The separate landing page on readthedocs.org will also be moving.

Reviewer: ```suggestion
- 1️⃣️ We are going to have 1 primary marketing website in the future,
while still running two different product for our Community & Business users.
  The first change that is now done is redirecting logged out users on `https://readthedocs.com/` to `about.readthedocs.com <https://about.readthedocs.com>`__.
  We will soon be doing this for logged out users going to `https://readthedocs.org/`,
  but not it's project-specific subpages.
```

Seems like an area where your suggestion to be very explicit is useful. I think we should probably have a link here explaining this process.

Reviewer 2: We have a blog post now that we can link 😉

Author: Yes, haha.. luxury problem: we'll need to resolve the order of how these are published. Ideally, we get the other blog post out first so it can be referenced here... but I think that the other blog post will have its own spotlight (mailing list + twitter post) and maybe even have more clicks than the newsletter, so it's not too important :)

Reviewer: Yea, we shouldn't block this post on that post

- 📚️ Documentation refactor underway:

>   A regular reader of our :doc:`user docs <readthedocs:index>` might notice that our navigation has been greatly changed and now contains only the `4 Diátaxis categories <https://diataxis.fr/>`__.

Reviewer: This seems odd... we should explain more about this work?

Author: Moved to the Upcoming Work section.

> - 🧹️ We are now able to delist projects from search engines more efficiently.

Reviewer: ```suggestion
- 🧹️ We are now able to delist unofficial projects from search engines.
```

  If you find an unofficial and unmaintained project,
  :doc:`please see our policy on unofficial and unmaintained projects <readthedocs:unofficial-projects>`.
- 🐞 Custom 404s with Sphinx DirHTML builder `are now supported <https://github.com/readthedocs/sphinx-notfound-page/issues/215>`__.

> - 🐞 :doc:`Pull Request <readthedocs:guides/pull-requests>` builds now link directly to the preview version of the documentation changes.

Author: Hmmmmmmm :)

The link isn't working on the latest build of *this* PR. Ping @ericholscher

Reviewer: Yea, it's not getting deployed til Tuesday. I don't think this is worth mentioning, since it's a bug fix, not a new feature.

Author: Updated the text... can decide if this is too insignificant or good marketing to make people aware that the behavior is unintentional and being fixed? It has an emoji that makes it easy to jump past for readers that aren't interested in our bug fixes.

> - 🐞 A brief occurrence of redirect loops around February 13-14 `was resolved <https://github.com/readthedocs/readthedocs.org/issues/10017>`__.

Reviewer: This seems way too small to mention. If we want to mention this kind of stuff, we should have a `Bugs fixed` or similar section, not in new features.

> - 🔒️ Vulnerability fixed: Pull requests preview builds `GHSA-h4cf-8gv8-4chf <https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-h4cf-8gv8-4chf>`__

Reviewer: We should use the official title on the security advisories, instead of writing custom things that aren't as specific.

> - 🔒️ Vulnerability fixed: Access to files from any project possible with crafted path traversal, see: `GHSA-5w8m-r7jm-mhp9 <https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-5w8m-r7jm-mhp9>`__.

Reviewer 3: This one was already included in the last month newsletter

Author: Oh snap! I wasn't looking carefully enough here. Thanks!

- 🔒️ Vulnerability fixed: Access to files from any project possible with crafted path traversal, see: `GHSA-5w8m-r7jm-mhp9 <https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-5w8m-r7jm-mhp9>`__.

You can always see the latest changes to our platforms in our :doc:`Read the Docs Changelog <readthedocs:changelog>`.

.. _rclone: https://rclone.org/

> Upcoming features

Reviewer: Is this section useful? I wonder if we should get rid of it..

-----------------

..

>   Notes:

Reviewer: These notes are really distracting when reviewing... We should remove them and just have notes in a README or template, I think? Not in the final copy.

  Next newsletter:
  Make a general announcement of our Roadmap

  General:

  When creating newsletter drafts, we keep the items here from the previous newsletter.
  This is in order to ensure due follow-up on features that are announced publicly.
  
  Feature done? A great follow-up is to add what was previously an upcoming feature as a released feature in the former section.
  
  Feature not done?
  Make sure that upcoming features are announced with a link to issues or PRs where the progress can be seen.
  If this is done, then subsequent newsletters aren't compelled to share progress when it's uninteresting.
  
  If a feature was announced as upcoming but isn't yet released,
  then try rephrasing the announcement as a general news update about the progress and where it can be followed.

- 🪄️ The Django application that handles uncached requests on all our hosted documentation is called El Proxito.
  And that application is getting a major refactor,
  improving both speed and features.
  Most of the refactoring work is already merged and within the coming weeks,
  we will start to roll it out as a kind of A/B testing.

> - ⏳️ The first work from our Diátaxis documentation refactor is now live and the general work has moved into shorter iterations.

Reviewer: Seems we're mentioning this twice? The above mention feels short, so maybe remove it, and just keep this longer explanation?

Author: Consolidated with the other item

  Once we have gathered all the final experience from the refactor,
  we will announce it together with practical inputs for how other documentation projects might approach such a refactor.
  If you are planning a refactor of a large documentation,
  we'd love to hear what your questions and concerns are and address those future blog posts and talks.

> - 🎬️ We're slowly moving towards a public beta of the whole Read the Docs dashboard experience. Here's a screenshot of what's to come...

Reviewer: ```suggestion
- 🎬️ We're slowly moving towards a public beta of our v2 dashboard experience. Here's a screenshot of what's to come...
```

.. image:: img/screenshot-dashboard-beta.png
   :align: center
   :scale: 50%

> Interested in all the details? `View our full Roadmap 📍️`_

Reviewer: ```suggestion
Want to follow along with our development progress? `View our full Roadmap 📍️`_
```

See our chosen highlights from Stack's documentation in the following `Twitter thread <https://twitter.com/readthedocs/status/1603095976117522433>`_:

.. raw:: html

   <blockquote class="twitter-tweet"><p lang="en" dir="ltr">GeoPandas is an open source project to make working with <a href="https://twitter.com/hashtag/geospatial?src=hash&amp;ref_src=twsrc%5Etfw">#geospatial</a> data in <a href="https://twitter.com/hashtag/Python?src=hash&amp;ref_src=twsrc%5Etfw">#Python</a> easier. <a href="https://twitter.com/geopandas?ref_src=twsrc%5Etfw">@GeoPandas</a> extends the datatypes used by pandas to allow spatial operations on geometric types.<br><br>We want to highlight some things we love from their docs.<br><br>🤏 (small) 🧵 <a href="https://t.co/Hj82s6SDQP">pic.twitter.com/Hj82s6SDQP</a></p>&mdash; Read the Docs (@readthedocs) <a href="https://twitter.com/readthedocs/status/1603095976117522433?ref_src=twsrc%5Etfw">December 14, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

Tip of the month
----------------

When you post links to your documentation on chat and social media,

> you will probably see a very uninspiring and generic preview.

Reviewer: ```suggestion
you will probably see a generic preview.
```

`very uninspiring` is a bit much :)

Enter `sphinxext-opengraph <https://github.com/wpilibsuite/sphinxext-opengraph>`__!

The extension allows you to configure your own preview card,
compatible with all major chat and social media platforms.
The feature which we really enjoy is that it will detect and use the first illustration used in your documentation.
If you want to see it in action,

> try sharing a link to this newsletter blog post (marketing unintended).

Reviewer: We should be promoting this more. Not sure the best avenue, but it's a great extension. 

We should probably note our official docs are using it above in the new features ^^

Author: Added a link to the user docs as well!
